test/parse_test.rb

```diff
@@ -141,7 +141,7 @@ def test_parse_takes_file_path
       file_contents = File.read(filepath, binmode: true, external_encoding: Encoding::UTF_8)
 
       file_contents.split(/(?<=\S)\n\n(?=\S)/).each do |snippet|
-        snippet = snippet.rstrip
+        snippet = snippet.rstrip + "\n"
```

Comment:
This is unfortunately a super specific use case of we copy the files so that we can check for out of bounds reads with valgrind. We've caught a couple of things this way, so I think I don't want to add a newline here just in case it helps us catch things in the the future. 

Could you explicitly skip the newline terminated file you gave so that we can keep this behavior?

Reply:
Sure.